fix(discover): Non Team projects weren't saveable

wmak opened the pull request:
- If a user wasn't in a team they couldn't save discover queries with
  projects that still belong to the organizaion

wmak committed:
fix(discover): Non Team projects weren't saveable

- If a user wasn't in a team they couldn't save discover queries with
  projects that still belong to the organizaion

markstory commented on src/sentry/discover/endpoints/serializers.py:
```diff
@@ -177,7 +177,7 @@ def validate_projects(self, projects):
 
         # Params will have project_ids that have been validated
-        if projects != set(self.context["params"]["project_id"]):
+        if not projects.intersection(set(self.context["params"]["project_id"])):
```
Will this still fail if projects from outside the organization are requested to be added to the query?

wmak replied:
Yep `test_save_with_wrong_projects` checks this, in the case the query is `project:some_project_from_another_org`

`projects` here would still have the other org's project but
`self.context["params"]["project_id"]` will only have projects for the user's org

markstory replied:
Yes the half valid, half invalid scenario is the one I was thinking of.

wmak replied:
🙏 Thanks, updated to avoid allowing that case

wmak committed:
ref: Adding another test, improving comment

wmak committed:
fix: Adding test for mixing valid/invalid

- If there are projects in the query that aren't in the validated set of
  projects then we should deny permission
